src/componentize.js

```diff
@@ -181,42 +183,80 @@ export async function componentize(jsSource, witWorld, opts) {
     console.log('--- Wizer Env ---');
     console.log(env);
   }
-
-  try {
-    let wizerProcess = spawnSync(
-      wizer,
-      [
-        '--allow-wasi',
-        '--init-func',
-        'componentize.wizer',
-        `--dir=${maybeWindowsPath(sourceDir)}`,
-        `--wasm-bulk-memory=true`,
-        '--inherit-env=true',
-        `-o=${output}`,
-        input,
-      ],
-      {
-        stdio: [null, stdout, stderr],
-        env,
-        input: maybeWindowsPath(
-          join(sourceDir, sourceName.slice(0, -3) + '.bindings.js')
-        ),
-        shell: true,
-        encoding: 'utf-8',
+  if (opts.enableAot) {
+    const wevalBin = await getWeval();
+
+    try {
+      let wevalProcess = spawnSync(
+        wevalBin,
+        [
+          'weval',
+          `--cache-ro ${aotCache}`,
+          `--dir ${maybeWindowsPath(sourceDir)}`,
+          '-w',
+          `-i ${input}`,
+          `-o ${output}`
+        ],
+        {
+          stdio: [null, stdout, stderr],
+          env,
+          input: maybeWindowsPath(
+            join(sourceDir, sourceName.slice(0, -3) + '.bindings.js')
+          ),
+          shell: true,
+          encoding: 'utf-8',
+        }
+      );
+      if (wevalProcess.status !== 0)
+        throw new Error('Wevaling failed to complete');
+    } catch (error) {
+      let err =
+        `Failed to initialize the compiled Wasm binary with Weval:\n` +
+        error.message;
+      if (DEBUG_BINDINGS) {
+        err += `\nBinary and sources available for debugging at ${tmpDir}\n`;
+      } else {
+        rmSync(tmpDir, { recursive: true });
       }
-    );
-    if (wizerProcess.status !== 0)
-      throw new Error('Wizering failed to complete');
-  } catch (error) {
-    let err =
-      `Failed to initialize the compiled Wasm binary with Wizer:\n` +
-      error.message;
-    if (DEBUG_BINDINGS) {
-      err += `\nBinary and sources available for debugging at ${tmpDir}\n`;
-    } else {
-      rmSync(tmpDir, { recursive: true });
+      throw new Error(err);
+    }
+  } else {
+    try {
+      let wizerProcess = spawnSync(
+        wizer,
+        [
+          '--allow-wasi',
```

Comment:
`--allow-wasi` must be enabled I believe as well.

Reply:
Looks like this. is not allowed as well.

```
error: Found argument '--allow-wasi' which wasn't expected, or isn't valid in this context
```

Reply:
Apparently this is on-by-default in Weval.